Closes #2595 deprecation of previous "superseded" functions

bms63 opened the pull request:
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

bms63 commented:
Question:  How to set it up so I can use the admiraldev branch with the new deprecation function?

bundfussr commented on R/derive_var_dthcaus.R:
Lines 285/286 need to be updated.

bundfussr commented:
> @bundfussr Question: How to set it up so I can use the admiraldev branch with the new deprecation function?

For the workflows which use staged dependencies it should work with the branch naming you are using. For the snapshot workflows it will not work. We would need to update these workflows to enable it. However, I'm not sure if it is worth the effort.

bms63 commented:
> > @bundfussr Question: How to set it up so I can use the admiraldev branch with the new deprecation function?
> 
> For the workflows which use staged dependencies it should work with the branch naming you are using. For the snapshot workflows it will not work. We would need to update these workflows to enable it. However, I'm not sure if it is worth the effort.

awesome thanks for confirming.  wnated to make sure this was set up correctly before i started to update

bms63 commented:
@ddsjoberg i am missing some minor detail here.  if i change to warn or error it works, but with new function `deprecate_inform` nothing is sent out. 

I installed my branch from admiraldev and can see the function available.  Any ideas?

![image](https://github.com/user-attachments/assets/30a6895d-34f3-4fc2-a5c8-ed5c2da9a0ce)

bms63 commented:
Ha - was thinking environments was being an issue...read the documentation and added these on to the function and it worked!!  But could not explain it.  Also, not grabbing the package name.  ummm...

![image](https://github.com/user-attachments/assets/1ef1fdae-4909-461a-b368-53481dd05db9)

bms63 commented:
Was also thinking about the standard message being issued and if we wanted/could change that for `deprecate_inform()`??  Right now it has warning in it.

ddsjoberg commented:
> Ha - was thinking environments was being an issue...read the documentation and added these on to the function and it worked!! But could not explain it. Also, not grabbing the package name. ummm...
> 

very good point! we should make these env args the default in `deprecate_inform()`

bms63 commented:
> > Ha - was thinking environments was being an issue...read the documentation and added these on to the function and it worked!! But could not explain it. Also, not grabbing the package name. ummm...
> 
> very good point! we should make these env args the default in `deprecate_inform()`

so interesting...moved arguments over to admiraldev and works now

![image](https://github.com/user-attachments/assets/caab77c2-48c2-4062-9eb4-b93346b6dd98)

bms63 commented:
Alright! Adventure is nearing the end. @ddsjoberg @manciniedoardo @bundfussr hoping you all can review please

ddsjoberg commented on R/derive_param_extreme_record.R:
```diff
@@ -136,6 +136,16 @@ derive_param_extreme_record <- function(dataset = NULL,
+    what = "derive_param_extreme_record()",
+    with = "derive_extreme_event()",
+    details = c(
+      x = "This message will turn into a warning with release of 1.3.0",
+      i = "https://pharmaverse.github.io/admiral/reference/derive_extreme_event.html"
```
I think I may have made this suggestion previously, but I wouldn't promise to the users the precise version when we're going to make deprecation message changes. We have a long deprecation cycle, and who knows if we need to change things in the future, e.g. we release v1.3 earlier than we had previously planned, or we wait longer. I recommend that before each release (and not just after a release), we search for all the deprecation messages in our package and update based on the deprecation calendar.

bms63 replied:
You did!  ummm.... maybe just "This message will turn into a warning at the next release"

I just want to stress to the user to start looking at updating their code.

ddsjoberg replied:
That could work, but also has a similar issue where if we promise the next release, but then the release timing changes for whatever reason. Also, we could decide to change our deprecation cycle.

My general advice is to not promise the users anything we don't have to. We have lengthy long deprecation cycle, which gives users literally years to update their code. I think the default messaging is sufficient. But, if you don't like it, you do you!!! 🕺🏼

bms63 replied:
Good point - my kid will be in high school by the time we remove these current functions form admiral!  

I am going to remove the 1.3 but keep the message that it will turn into a warning for a future release.

bms63 commented:
Hi @bundfussr in your upcoming review.  I removed the `deprecate_inform` for `date_source`.  It was causing some weird issues in the `derive_var_extreme_dtm()` tests calls as it called many times.

Since date_source is used in `derive_var_extreme_dtm()` I figured user will already be aware of its deprecation and we don't need to overkill it here.  

If you see a quick fix, then I can implement otherwise I think we can move on!

bundfussr commented on tests/testthat/test-get_summary_records.R:
```diff
@@ -22,6 +22,46 @@ test_that("get_summary_records Test 1: Summarize average of triplicate ECG inter
     "XYZ-1002", 9,      "QTcF Int. (msec)", "Visit 3",  "2016-03-24T10:56", 402,   "Active 20mg"
   )
 
+  expect_snapshot(
```
The title of this test should be updated to "deprecation message if function is called" and the tests below should be renumbered.

bms63 replied:
this is updated

bundfussr commented on R/derive_var_dthcaus.R:
```diff
@@ -317,6 +327,16 @@ dthcaus_source <- function(dataset_name,
                            set_values_to = NULL) {
+  deprecate_inform(
+    when = "1.2.0",
+    what = "date_source()",
```
```suggestion
    what = "dthcaus_source()",
```

bundfussr commented on R/derive_var_dthcaus.R:
```diff
@@ -247,8 +257,8 @@ derive_var_dthcaus <- function(dataset,
-#' function and `dthcaus_source()` have been superseded in favor of
+#' `r lifecycle::badge("deprecated")` The `derive_var_dthcaus()`
+#' function and `dthcaus_source()` have been deprecated in favor of
 #' `derive_vars_extreme_event()`.
```
```suggestion
#' `event()`.
```

bundfussr commented on R/derive_var_extreme_date.R:
```diff
@@ -575,8 +596,8 @@ derive_var_extreme_dt <- function(dataset,
-#' `r lifecycle::badge("superseded")` The `date_source()`
-#' function has been superseded in favor of `derive_vars_extreme_event()`.
+#' `r lifecycle::badge("deprecated")` The `date_source()`
+#' function has been deprecated in favor of `derive_vars_extreme_event()`.
```
```suggestion
#' function has been deprecated in favor of `event()`.
```

bms63 commented on R/derive_param_extreme_record.R:
```diff
@@ -136,6 +136,16 @@ derive_param_extreme_record <- function(dataset = NULL,
+    with = "derive_extreme_event()",
+    details = c(
+      x = "This message will turn into a warning in the next release.",
+      i = "https://pharmaverse.github.io/admiral/reference/derive_extreme_event.html"
```
I think we should link the function documentation in the message.  I can make a note of this in the guidance document.

bundfussr replied:
Hm, linking to the deprecation guidance (either in the admiral reference or the programming strategy) was a good idea because most likely many users are not aware of it.

bundfussr replied:
If we want to link to the function documentation, I would change the order such that we get
```
#> `get_summary_records()` was deprecated in admiral 1.2.0.
#> ℹ Please use `derive_summary_records()` instead.
#> ℹ https://pharmaverse.github.io/admiral/reference/derive_summary_records.html
#> ✖ This message will turn into a warning in the next release.
```
instead of
```
#> `get_summary_records()` was deprecated in admiral 1.2.0.
#> ℹ Please use `derive_summary_records()` instead.
#> ✖ This message will turn into a warning in the next release.
#> ℹ https://pharmaverse.github.io/admiral/reference/derive_summary_records.html
```

bms63 replied:
ah okay - i can make the change.

bundfussr commented on R/derive_var_dthcaus.R:
```diff
@@ -247,8 +257,8 @@ derive_var_dthcaus <- function(dataset,
 #' @description
-#' `r lifecycle::badge("superseded")` The `derive_var_dthcaus()`
-#' function and `dthcaus_source()` have been superseded in favor of
+#' `r lifecycle::badge("deprecated")` The `derive_var_dthcaus()`
```
```suggestion
#' `r lifecycle::badge("deprecated")` The `dthcaus_source()`
```

bundfussr commented:
> Hi @bundfussr in your upcoming review. I removed the `deprecate_inform` for `date_source`. It was causing some weird messaging issues in the `derive_var_extreme_dtm()` tests calls as it called many times.
> 
> Since date_source is used in `derive_var_extreme_dtm()` I figured user will already be aware of its deprecation and we don't need to overkill it here.
> 
> If you see a quick fix, then I can implement otherwise I think we can move on!

I think if you include the creation of the `date_source()` objects in the `expect_snapshot()` call, it should work. For example:
```
test_that("derive_var_extreme_dtm Test 5: Message sent to users", {
  expect_snapshot({
    ae_start <- date_source(
      dataset_name = "ae",
      date = convert_dtc_to_dtm(AESTDTC),
      set_values_to = exprs(
        LALVDOM = "AE",
        LALVSEQ = AESEQ,
        LALVVAR = "AESTDTC"
      )
    )

    ae_end <- date_source(
      dataset_name = "ae",
      date = AEENDTM,
      set_values_to = exprs(
        LALVDOM = "AE",
        LALVSEQ = AESEQ,
        LALVVAR = "AEENDTC"
      )
    )

    adsl_trtdate <- date_source(
      dataset_name = "adsl",
      date = TRTEDTM,
      set_values_to = exprs(
        LALVDOM = "ADSL",
        LALVSEQ = NA_integer_,
        LALVVAR = "TRTEDTM"
      )
    )

    adsl_dthdate <- date_source(
      dataset_name = "adsl",
      date = DTHDT,
      filter = nchar(DTHDTC) >= 10,
      set_values_to = exprs(
        LALVDOM = "ADSL",
        LALVSEQ = NA_integer_,
        LALVVAR = "DTHDTC"
      )
    )

    derive_var_extreme_dtm(
      adsl,
      new_var = LSTALVDTM,
      source_datasets = list(ae = ae, adsl = adsl),
      ae_start, ae_end, adsl_trtdate, adsl_dthdate,
      mode = "last"
    )
  })
})
```

bms63 commented on NEWS.md:
```diff
@@ -59,13 +59,23 @@ default. To enable it the new admiral option `save_memory` has to be set to
+- The following function arguments are entering the next phase of the deprecation process: (#2487) (#2595)
 
-### Phase 1 (warning)
+### Phase 1 (message)
+  
+  - `derive_param_extreme_record()` is deprecated and replaced by `derive_extreme_event()`
+  - `derive_var_dthcaus()` is deprecated and replaced by `derive_vars_extreme_event()`
```
Updated

bms63 commented on R/user_helpers.R:
```diff
@@ -60,11 +60,11 @@ use_ad_template <- function(adam_name = "adsl",
   }
 
-  invisible(TRUE)
+  invisible(TRUE) # nocov
```
i wanted to try this out and see if we get this line removed from the code coverage.  was thinking other functions that are not critical to users or experimental could have this added in temporarily or for the long haul

...pushing for that sweet 100% but we can never get there cause this line can not be tested since it is interactive

bundfussr replied:
Only line 64
```
  file.edit(save_path)
```
can't be tested because it is interactive.

Line 63 and 67 can be tested and are already covered. Thus I would remove the `# nocov` comment from these lines. I think we should add `# nocov` only were it is absolutely necessary. If we claim 100% coverage but have a lot of `# nocov` comments, it doesn't look convincing.

bms63 commented:
> > Hi @bundfussr in your upcoming review. I removed the `deprecate_inform` for `date_source`. It was causing some weird messaging issues in the `derive_var_extreme_dtm()` tests calls as it called many times.
> > Since date_source is used in `derive_var_extreme_dtm()` I figured user will already be aware of its deprecation and we don't need to overkill it here.
> > If you see a quick fix, then I can implement otherwise I think we can move on!
> 
> I think if you include the creation of the `date_source()` objects in the `expect_snapshot()` call, it should work. For example:
> 
> ```
> test_that("derive_var_extreme_dtm Test 5: Message sent to users", {
>   expect_snapshot({
>     ae_start <- date_source(
>       dataset_name = "ae",
>       date = convert_dtc_to_dtm(AESTDTC),
>       set_values_to = exprs(
>         LALVDOM = "AE",
>         LALVSEQ = AESEQ,
>         LALVVAR = "AESTDTC"
>       )
>     )
> 
>     ae_end <- date_source(
>       dataset_name = "ae",
>       date = AEENDTM,
>       set_values_to = exprs(
>         LALVDOM = "AE",
>         LALVSEQ = AESEQ,
>         LALVVAR = "AEENDTC"
>       )
>     )
> 
>     adsl_trtdate <- date_source(
>       dataset_name = "adsl",
>       date = TRTEDTM,
>       set_values_to = exprs(
>         LALVDOM = "ADSL",
>         LALVSEQ = NA_integer_,
>         LALVVAR = "TRTEDTM"
>       )
>     )
> 
>     adsl_dthdate <- date_source(
>       dataset_name = "adsl",
>       date = DTHDT,
>       filter = nchar(DTHDTC) >= 10,
>       set_values_to = exprs(
>         LALVDOM = "ADSL",
>         LALVSEQ = NA_integer_,
>         LALVVAR = "DTHDTC"
>       )
>     )
> 
>     derive_var_extreme_dtm(
>       adsl,
>       new_var = LSTALVDTM,
>       source_datasets = list(ae = ae, adsl = adsl),
>       ae_start, ae_end, adsl_trtdate, adsl_dthdate,
>       mode = "last"
>     )
>   })
> })
> ```

that worked - tips is to do `expect_snapshot({})` rather than `expect_snapshot()`  when multiple objects, functions are being created/used

bms63 commented:
@bundfussr updates completed

bms63 commented:
I will make updates to guidance in admiraldev later today!

bundfussr commented on NEWS.md:
```diff
@@ -59,13 +59,24 @@ default. To enable it the new admiral option `save_memory` has to be set to
+- The following function arguments are entering the next phase of the [deprecation process](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation): (#2487) (#2595)
 
-### Phase 1 (warning)
+**Phase 1 (message)**
+  
+  - `derive_param_extreme_record()` is deprecated and replaced by `derive_extreme_event()`
+  - `derive_var_dthcaus()` is deprecated and replaced by `derive_vars_extreme_event()`
+  - `date_source()` is deprecated and replaced by `event()`
+  - `dthcaus_source()` is deprecated and replaced by `event()`
+  - `derive_var_extreme_dt()` and `derive_var_extreme_dtm()` are deprecated and replaced by `derive_vars_extreme_event()`
+  - `get_summary_records()` is deprecated. Please use `derive_summary_records()` with the `dataset_add` argument and without the `dataset` argument.
+
+  
+**Phase 2 (warning)**
+
+**Phase 3 (error)**
```
It should be indented by four spaces to look nice.
```suggestion
    **Phase 1 (message)**
  
    - `derive_param_extreme_record()` is deprecated and replaced by `derive_extreme_event()`
    - `derive_var_dthcaus()` is deprecated and replaced by `derive_vars_extreme_event()`
    - `date_source()` is deprecated and replaced by `event()`
    - `dthcaus_source()` is deprecated and replaced by `event()`
    - `derive_var_extreme_dt()` and `derive_var_extreme_dtm()` are deprecated and replaced by `derive_vars_extreme_event()`
    - `get_summary_records()` is deprecated. Please use `derive_summary_records()` with the `dataset_add` argument and without the `dataset` argument.
  
    **Phase 2 (warning)**

    **Phase 3 (error)**
```
Should we remove "Phase 2" and "Phase 3"?

bms63 replied:
Indented.

I like to keep those sections as placeholders so we keep things identical between releases.  I put in some text to be explicit on why they are empty.

bundfussr commented on R/derive_param_extreme_record.R:
```diff
@@ -136,6 +136,17 @@ derive_param_extreme_record <- function(dataset = NULL,
+    what = "derive_param_extreme_record()",
+    with = "derive_extreme_event()",
+    details = c(
+      x = "This message will turn into a warning for at least one year.",
```
I think this message is misleading. If a user calls the function in December 2025, it looks as if the warning will not occur before end of 2026 although we will change it at the beginning of 2026.

What about:
```suggestion
      x = "This message will turn into a warning at the beginning of 2026.",
```
